feat: output more info if up to date

nichmor opened the pull request:
Right now when doing `pixi global install something` on an already up-to-date environment, it will just print that the env is already up-to-date.

This PR introduces a more detailed view of up-to-date environment, which uses the same format as pixi global list

![image](https://github.com/user-attachments/assets/4c7351b6-2518-4193-9f9e-aca624943b0d)


some other niceties 🍪 :

* It also move code from cli/global under separate module
* I've added `test-specific-test` that I think it is handy to have when you want to test only one test without running all of them. Also running without `--numprocesses=auto` make debugger hook work.
* I also added output structure for `verify_cli_command`, so it is more easier to debug what subprocess returned rather than going inside `verify_cli_command`



upd:

after a discussion with @Hofer-Julian we decided to use `always` tree output after installing with some information in there
this is how it looks with new colouring and styling:

![image](https://github.com/user-attachments/assets/7f4f9e12-631c-4533-a06f-f87cc569c54b)

Hofer-Julian commented on src/global/install.rs:
```diff
@@ -99,7 +99,7 @@ fn get_catch_all_arg(shell: &ShellEnum) -> &str {
 /// Create the executable scripts by modifying the activation script
 /// to activate the environment and run the executable.
-pub(crate) async fn create_executable_scripts(
+pub(crate) async fn create_executable_scripts<'a>(
```
Is this really necessary?

Hofer-Julian commented on tests/integration/common.py:
```diff
@@ -12,6 +12,22 @@ class ExitCode(IntEnum):
     INCORRECT_USAGE = 2
 
 
+class Output:
+    command: list[Path | str]
+    stdout: str
+    stderr: str
+    returncode: int
+
+    def __init__(self, command: list[Path | str], stdout: str, stderr: str, returncode: int):
+        self.command = command
+        self.stdout = stdout
+        self.stderr = stderr
+        self.returncode = returncode
+
+    def __str__(self) -> str:
+        return f"command: {self.command}"
```
I don't particularly mind, but we don't use this yet, right?

nichmor replied:
We don't use it directly , but I think it is handy to have while debugging the test to check what is the output ( which I've personally have done in this PR ) that's why I think it would be nice to have in our dev toolchain

Hofer-Julian commented on src/cli/global/install.rs:
```diff
@@ -116,7 +119,23 @@ pub async fn execute(args: Args) -> miette::Result<()> {
     }
-    state_changes.report();
+    // When the environment is already up to date, we want to show a more detailed view about it.
+    if !state_changes.has_changed() {
```
If certain environments contain no changes, but others do we should also print `environments up-to-date`.

Like here I would have expected, `Environment rattler-build is already up to date
![image](https://github.com/user-attachments/assets/c05ab727-b850-414e-a2bd-76c00b0f3a1a)

Hofer-Julian reviewed (commented):
Looks great!

nichmor commented:
after a discussion with @Hofer-Julian  we decided to use tree output with some information in there

this is the new output:

![image](https://github.com/user-attachments/assets/9cd364e3-a7f0-40bb-bb3e-729d8f36ac7c)

ruben-arts commented:
![image](https://github.com/user-attachments/assets/9ee0e032-e4d3-4971-bd05-ccfe119f440f)
Why doe it give the already installed on the simple version and not on the versioned one?

nichmor commented:
> ![image](https://private-user-images.githubusercontent.com/12893423/377777052-9ee0e032-e4d3-4971-bd05-ccfe119f440f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNDEzMTQsIm5iZiI6MTcyOTI0MTAxNCwicGF0aCI6Ii8xMjg5MzQyMy8zNzc3NzcwNTItOWVlMGUwMzItZTRkMy00OTcxLWJkMDUtY2NmZTExOWY0NDBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDA4NDMzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmYTMxOWRmNTI1NWYyYjA2NjBmNWY1MTUwYTVlMTUzZmM5MWFlYWY3YmExMGQ4MjRmZGFhODQ5YmJmOTQ2OTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HjikMvFSpxtjWCAU-J-O94xmZPE-bDzcZDoANVcSxrA) Why doe it give the already installed on the simple version and not on the versioned one?


Fixed all 3 situations ( so it also output exposes correctly ) and also added a test:

![Screenshot 2024-10-18 at 11 44 53](https://github.com/user-attachments/assets/f1474a2f-0b8c-468c-8eca-269d0b892757)

![Screenshot 2024-10-18 at 11 45 12](https://github.com/user-attachments/assets/d5b13e58-d038-4938-b2a5-2b4e383eb1de)


![Screenshot 2024-10-18 at 11 45 43](https://github.com/user-attachments/assets/105f5b95-0878-4236-8bc8-9882a59277f3)

Hofer-Julian reviewed (approved):
Awesome work 🤩
